src/server/scripts/Outland/CoilfangReservoir/TheSlavePens/boss_ahune.cpp

```diff
@@ -0,0 +1,775 @@
+                    instance->SetGuidData(DATA_FLAMECALLER_001, flameCaller001->GetGUID());
+                if (TempSummon* flameCaller002 = me->SummonCreature(NPC_EARTHEN_RING_FLAMECALLER, SummonPositions[7], TEMPSUMMON_MANUAL_DESPAWN))
+                    instance->SetGuidData(DATA_FLAMECALLER_002, flameCaller002->GetGUID());
+
+                _events.SetPhase(PHASE_ONE);
+                _events.ScheduleEvent(EVENT_SUMMON_COLDWEAVE, 8000, 0, PHASE_ONE);
+                _events.ScheduleEvent(EVENT_SUMMON_HAILSTONE, 6000, 0, PHASE_ONE);
+                _events.ScheduleEvent(EVENT_SUMMON_AHUNE, 5000);
+
+            }
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            if (!UpdateVictim())
+                return;
+
+            _events.Update(diff);
+
+            while (uint32 eventId = _events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                case EVENT_SUMMON_HAILSTONE:
+                    DoCast(SPELL_SUMMON_HAILSTONE);
+                    break;
+                case EVENT_SUMMON_COLDWEAVE:
+                    me->CastSpell(me, SPELL_SUMMON_COLDWEAVE);
+                    me->CastSpell(me, SPELL_SUMMON_COLDWEAVE);
+                    _events.ScheduleEvent(EVENT_SUMMON_COLDWEAVE, 5000, 0, PHASE_ONE);
+                    if (_submerged)
+                        me->CastSpell(me, SPELL_SUMMON_FROSTWIND);
+                    _events.ScheduleEvent(EVENT_SUMMON_FROSTWIND, 7000, 0, PHASE_ONE);
+                    break;
+                case EVENT_SUMMON_AHUNE:
+                    if (TempSummon* ahune = me->SummonCreature(NPC_AHUNE, SummonPositions[0], TEMPSUMMON_DEAD_DESPAWN))
+                    {
+                        ahune->SummonCreature(NPC_FROZEN_CORE, SummonPositions[1], TEMPSUMMON_CORPSE_DESPAWN);
+                        me->CastSpell(ahune, SPELL_RESURFACE);
+                    }
+                    break;
+                }
+            }
+
+        }
+    };
+
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return new npc_ahune_bunnyAI(creature);
+    }
+};
+
+class go_ahune_ice_stone : public GameObjectScript
+{
+public:
+    go_ahune_ice_stone() : GameObjectScript("go_ahune_ice_stone") { }
+
+    bool OnGossipSelect(Player* player, GameObject* go, uint32 /*sender*/, uint32 /*action*/)
+    {
+        InstanceScript* instance = go->GetInstanceScript();
+        player->PlayerTalkClass->ClearMenus();
+
+        if (Creature* ahuneBunny = go->SummonCreature(NPC_AHUNE_LOC_BUNNY, SummonPositions[0], TEMPSUMMON_CORPSE_DESPAWN))
+        {
+            ahuneBunny->AI()->DoAction(ACTION_START_EVENT);
+            ahuneBunny->SetInCombatWithZone();
+        }
+        if (Creature* luma = ObjectAccessor::GetCreature(*go, instance->GetGuidData(DATA_LUMA_SKYMOTHER)))
+        {
+            Unit* target = luma->SelectNearestPlayer(200.0f);
```

Comment:
The passed argument (Player* player) should be the target not the nearest player.